src/client/pythonEnvironments/discovery/subenv.ts

```
@@ -17,13 +19,13 @@ type ExecutableFinderFunc = (python: string) => Promise<string | undefined>;
  * @param finders - the functions specific to different Python environment types
  */
 export async function getName(python: string, finders: NameFinderFunc[]): Promise<string | undefined> {
-    for (const find of finders) {
-        const found = await find(python);
-        if (found && found !== '') {
-            return found;
-        }
+    try {
+        const envNames = await Promise.all(finders.map((find) => find(python)));
+
+        return envNames.find((name) => name && name !== '');
+    } catch {
+        return undefined;
```

Isn't this a breaking change.
Previously the code would throw an exception, now we're swallowing the exception!

---

I.e. if one of the functions `find` failed (promise rejected), then it would get handled further up, now this is handling the error and swallowing it

---

🤦‍♀️  will remove the try/catch, thank you